packages/core/src/helpers/color/color.ts

```diff
@@ -38,88 +38,38 @@ export type Palette =
   | WarningColor
   | ErrorColor;
 
-type NeutralColor =
-  | 'neutral-900'
-  | 'neutral-800'
-  | 'neutral-700'
-  | 'neutral-600'
-  | 'neutral-500'
-  | 'neutral-400'
-  | 'neutral-300'
-  | 'neutral-200'
-  | 'neutral-050'
-  | 'neutral-white'
-  | 'neutral-black';
```

Comment:
This all is much more difficult to read now...

What do you think if only like such would be a change:

```ts
type NeutralColor =   `neutral-${'900' | '800' | '700' | '600' | '500' | '400' | '300' | '200' | '050' | 'white' | 'black'}`;
```

Or use the `Weight` type, and exclude what's not needed for some cases?

Reply:
Fair enough. I reused types as most as I could, but it does make it slightly more annoying to read.

How does it look now? Want me to get rid of `Muted` and `Vivid` too?

Reply:
Don't mind it, though I feel it makes it a little less readable, take a look.

Reply:
So I feel like such we now have only one place where available weight for colors is defined, and others only extract or exclude what they need.

Reply:
Cool, unfortunately `Extract` does not suggest keys in the union so it's a bit iffy, but agree.
I'll merge like it is and we can always revisit it if we don't like it.